cxx-squid/src/main/java/org/sonar/cxx/preprocessor/ExpressionEvaluator.java

```diff
@@ -144,7 +145,8 @@ private BigInteger evalComplexAst(AstNode exprAst) {
     } else if ("functionlikeMacro".equals(nodeType)) {
       return evalFunctionlikeMacro(exprAst);
     } else {
-      throw new EvaluationException("Unknown expression type '" + nodeType + "'");
+      LOG.error("'evalComplexAst' Unknown expression type '" + nodeType + "' for AstExt '" + exprAst.getToken() + "', assuming 0");
+      return BigInteger.ZERO;
```

Comment:
Here I'm not sure. Isn't this an internal bug where we can't find the right node type? Means typically we have to extend the parser? Question is how "tolerant" the parser should be in such a case?